tests/compiler/test_basics.py

```diff
@@ -212,3 +218,116 @@ def test_value_set_twice(snippetcompiler):
+        """.strip()
+
+    # The model for the test module, regardless of value
+    test_module_model: str = f"value = {test_module}::get_value()"
+
+    # install main module (make it v1 so there are no restrictions on what it can depend on)
+    v1_module_from_template(
+        v1_template_path,
+        os.path.join(libs_dir, f"{main_module}"),
+        new_name=main_module,
+        new_content_init_cf=f"""
+import {test_module}
+
+# test variable import
+test_module_value = {test_module}::value
+# test plugin call
+test_module_value = {test_module}::get_value()
+# test intermodule Python imports
+test_module_value = {main_module}::get_test_module_value()
+        """.strip(),
+        new_content_init_py=f"""
+from inmanta.plugins import plugin
+from {const.PLUGINS_PACKAGE}.{test_module} import VALUE
+
+
+@plugin
+def get_test_module_value() -> "int":
+    return VALUE
+        """.strip(),
+    )
+
+    def verify_compile(expected_value: int) -> None:
+        """
+        Verify compilation by importing main module and checking its variable's value.
+        """
+        snippetcompiler_clean.setup_for_snippet(
+            f"""
+import {main_module}
+
+# make sure imported variable has expected value (these statements will produce compiler error if it doesn't)
+value = {main_module}::test_module_value
+value = {expected_value}
+            """.strip(),
+            add_to_module_path=[libs_dir],
+            # set autostd=False because no v2 std exists at the time of writing
+            autostd=False,
+            install_project=False,
+        )
+        compiler.do_compile()
+
+    # install test module as v1 and verify compile
+    v1_module_from_template(
+        v1_template_path,
+        os.path.join(libs_dir, test_module),
+        new_name=test_module,
+        new_content_init_cf=test_module_model,
+        new_content_init_py=test_module_plugin_contents(1),
+    )
+    verify_compile(1)
+
+    # install module as v2 (on top of v1) and verify compile
+    v2_module_path: str = os.path.join(str(tmpdir), test_module)
+    module_from_template(
+        v2_template_path,
+        v2_module_path,
+        new_name=test_module,
+        new_content_init_cf=test_module_model,
+        new_content_init_py=test_module_plugin_contents(2),
+        install=True,
+        editable=True,
+    )
+    verify_compile(2)
+
+    # verify editable mode for plugins
+    with open(os.path.join(v2_module_path, const.PLUGINS_PACKAGE, test_module, "__init__.py"), "w") as fh:
+        fh.write(test_module_plugin_contents(3))
+    verify_compile(3)
+
+    # verify editable mode for model
+    with open(os.path.join(v2_module_path, "model", "_init.cf"), "w") as fh:
+        fh.write("value = 4")
+    # can't just verify_compile(4) because the plugin is still at 3 and promoting it to 4 would not test changes in the model
+    with pytest.raises(DoubleSetException) as excinfo:
+        verify_compile(3)
+    assert excinfo.value.newvalue == 4
```

Comment:
Strictly speaking either the old or new value could be 4, but we can fix that if it every breaks

Reply:
Indeed, I was thinking the same. This should be stable unless we make explicit changes to the compile flow, or if there is some nondeterministic behavior we're not aware of. In the first case it's easy to fix, in the second case we'd want to make an attempt to make the behavior deterministic I think.